From a43fd59a4fc57b16c8646f3eff4ecc2b007ea5e9 Mon Sep 17 00:00:00 2001
From: Martin Hoffmann <martin.de@myself.com>
Date: Wed, 26 Jul 2017 12:10:37 +0200
Subject: [PATCH 6/6] audio: Fix WBS sample rate usage

Fixes: BUGBASH-302

Correctly use 16khz for the WBS-SCO pcm_config.
No more garbled or mute in-call sound on bt.

Apply together with a device specific port of:
https://review.lineageos.org/#/c/177996/
and REMOVE any definitions of
SCO_WBS_SAMPLE_RATE
https://review.lineageos.org/#/c/178371/

This requires testing for all devices using:
android_hardware_samsung
(universal audio HAL part)

Change-Id: I84fb1b234bf2880bf98bf99f1e4ff6c05500b0e3
---
 audio/voice.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/audio/voice.c b/audio/voice.c
index a4b6542..d058359 100644
--- a/audio/voice.c
+++ b/audio/voice.c
@@ -73,10 +73,9 @@ struct pcm_config pcm_config_voice_sco = {
     .format = PCM_FORMAT_S16_LE,
 };
 
-/* SCO WB and NB uses 8kHz for now, 16kHz it's on TO DO*/
 struct pcm_config pcm_config_voice_sco_wb = {
     .channels = 1,
-    .rate = SCO_DEFAULT_SAMPLING_RATE,
+    .rate = SCO_WB_SAMPLING_RATE,
     .period_size = SCO_PERIOD_SIZE,
     .period_count = SCO_PERIOD_COUNT,
     .format = PCM_FORMAT_S16_LE,
-- 
2.17.1

